### PR TITLE
Project ToDo list - 

### PR DESCRIPTION
**Pull request by expjazz**
### The project has all the features requested:

- You can create projects, and every project contains a list of ToDo items.
- You can edit items, delete, add, create, and delete projects.
- The project is divided into modules, uses ES6 
- You can save your projects in Local Storage

#### Looking forward on how to improve the projects!

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
+/* eslint-disable no-use-before-define */
+/* eslint-disable import/no-cycle */
```
Comment (Oghenebrume50) [inline src/helpers.js]: This is an indication that there is a circular dependency in your code, this is not good coding practice, please remove this rule and fix the problem, the affected file is the 'Todolistgenerator' 

[see this](https://stackoverflow.com/questions/51094117/dependency-cycle-detected-import-no-cycle)

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED

Hi @expjazz @mariosknl nice job done here, the project is coming along very well, :slightly_smiling_face:  I have highlighted a few things that need fixing. 

- Currently, I can make a today with wrong or blank data, for example putting "good boy' as the due date of a todo works perfectly, this should not be. 
Also, see the error highlighted  below 

**Please make changes and request for another review** :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask:

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress :+1: 
You have done well fixing issues raised by the previous reviewer. We however, need to fix a few things you missed to address. 
- [x] Inputting a random text as date when creating a project doesn't raise any errors. Make sure you validate the date input. Use HTML date to pick date values.
- [x] It's possible to create a task with empty fields. This should not be happening. Please fix that.

### Martin Afani

**Comment by expjazz** [comment]
### We made all the requested changes.

- Now we added validations for a valid date. Now you can only create a task or a project in the future.
- Now both Projects and Tasks cannot be valid if they are empty.
- There are no ways to put random text in the dates since we are using the inputs of type DateTime local.

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @expjazz 

Great job with modification from former TSE
todo2

### [changes required]

In the same continuation please try to also prevent the creation of 'EMPTY' projects 
That's the last thing left in my opinion
todo

Waiting for the last re-review hopefully

Happy coding

**Comment by expjazz** [comment]
## Dear TSE, 

#### We already implemented those validations. What I think happened is that the last TSE checked the wrong branch in his local machine (now we deleted that branch). The branch we wanted to merge and still want is the 'featureV1'. I just tested this feature on my end and it worked. 

image

**Review by shloch — APPROVED** [review]
### Good to go....i happened to have claimed this request again.......
As per our chat, see you on further projects
accomplished

Happy coding

**Pull request merged**